gnosis-buyback/contracts/Buyback.sol

```diff
@@ -0,0 +1,539 @@
+            total = total.add(_auctionAmounts[i]);
+            auctionIndexWithAmount[buybackId][_auctionIndexes[i]] = _auctionAmounts[i];
+        }
+
+        Buyback memory buyback = Buyback(
+            _userAddress,
+            _sellToken, 
+            _buyToken, 
+            _burnAddress,
+            _burn, 
+            _auctionIndexes, 
+            _tipAmount,
+            _expires,
+            false,
+            false
+        );
+
+        // map buyback id to the buyback config
+        buybacks.push(buyback);
+        // store buyback id for user
+        userBuybacks[_userAddress].push(buybackId);
+
+        // increase amount of tips and 
+        userBuybackAmountTotal[_userAddress][_sellToken] = userBuybackAmountTotal[_userAddress][_sellToken].add(total);
+        userTipAmountTotal[_userAddress] = userTipAmountTotal[_userAddress].add(_tipAmount);
+
+        emit AddBuyBack(
+            buybackId,
+            _userAddress,
+            _buyToken,
+            _sellToken, 
+            _burnAddress, 
+            _burn, 
+            _auctionIndexes, 
+            _auctionAmounts
+        );
+    }
+
+    /**
+     * @notice postSellOrder approve trading
+     * @param _buybackId buyback id
+     */
+    function postSellOrder(uint _buybackId) public {
+        Buyback storage userBuyback = buybacks[_buybackId];
+
+        uint[] memory auctionIndexes = userBuyback.auctionIndexes;
+        address userAddress = userBuyback.userAddress;
+        address sellToken = userBuyback.sellToken;
+        uint tipAmount = userBuyback.tipAmount;
+
+        require(auctionIndexes.length > 0, "buyback doesn't exist");
+        // ensure buyback has not  been executed
+        require(userBuyback.hasExecutedBuyback == false, "buyback has been executed");
+        // check if buyback has expired
+        require(isTimePassed(userBuyback.expires), "buyback has expired");
+        require(hasEnoughBalance(_buybackId, auctionIndexes, userAddress, sellToken), "user does not have enough balance to post sell order");
+        require(hasEnoughTippingBalance(userAddress, tipAmount), "user does not have enough tipping balance");
+
+
+        uint newBuyerBalance;
+        uint dxIndex;
+        uint totalAmount = 0;
+
+        address buyToken = userBuyback.buyToken;        
+        uint userSellTokenBalance = balances[userAddress][sellToken];
+
+        for( uint i = 0; i < auctionIndexes.length; i++ ) {
+            // get the auction amount for that index
+            uint amount = auctionIndexWithAmount[_buybackId][auctionIndexes[i]]; 
+            totalAmount = totalAmount.add(amount);
+
+            // check if the user as enough balance
+            require(userSellTokenBalance >= amount, "user does not have enough balance"); 
+
+            approveDutchX(sellToken, amount);
+            depositDutchx(sellToken, amount);
+
+            (dxIndex, newBuyerBalance) = dx.postSellOrder(sellToken, buyToken, auctionIndexes[i], amount);
```

Comment:
When we execute a buy back, we sell for ALL the auctions in the buy back?

For example:
```js
{
  ...
  _auctionIndexes: [35, 50]
}
```

If the current auction is 35, it'll try to sell in number 35 (that it's all right), but then It'll try to sell in the 50, and it'll revert.